.azure/scripts/setup-kubernetes.sh

```diff
@@ -78,47 +143,85 @@ if [ "$TEST_CLUSTER" = "minikube" ]; then
     fi
 
     minikube addons enable default-storageclass
+    add_docker_hub_credentials_to_kubernetes "$TEST_CLUSTER"
+    setup_registry_proxy
 
-    # Add Docker hub credentials to Minikube
-    if [ "$COPY_DOCKER_LOGIN" = "true" ]
-    then
-      set +ex
+elif [ "$TEST_CLUSTER" = "kind" ]; then
+    install_kubectl
+    install_kubernetes_provisioner
 
-      docker exec "minikube" bash -c "echo '$(cat $HOME/.docker/config.json)'| sudo tee -a /var/lib/kubelet/config.json > /dev/null && sudo systemctl restart kubelet"
+    setup_kube_directory
```

Comment:
Wouldn't be better to run the common parts before the if statement to avoid redundant lines?